thefuck/rules/docker_image_being_used_by_container.py

```diff
@@ -0,0 +1,20 @@
+from thefuck.utils import for_app
+from thefuck.shells import shell
+
+
+@for_app('docker')
+def match(command):
+    '''
+    Matches a command's output with docker's output
+    warning you that you need to remove a container before removing an image.
+    '''
+    return 'image is being used by running container' in command.output
+
+
+def get_new_command(command):
+    '''
+    Prepends docker container rm -f {container ID} to
+    the previous docker image rm {image ID} command
+    '''
+    container_id = command.output.strip().split(' ')
+    return shell.and_('docker container rm -f {} && {}').format(container_id[-1], command.script)
```

Comment:
The thing is, some shells – such as Fish<sup>1</sup> – do not support the `&&` operator. They use things like `; and` instead, for example. That's why each of the shells in TheFuck implement their own `AND` operator (i.e. `shell.and_()`. It takes a sequence of commands and joins them by the supported operator:

```python
oneliner = shell.and('cmd1', 'cmd2', 'cmd3')
```

So, please change it like the following:

```suggestion
    return shell.and_('docker container rm -f {}', '{}').format(container_id[-1], command.script)
```  
 
> * <sup>1</sup> Recent versions of Fish support `&&` but we must keep backward compatibility

Reply:
That makes sense! I'll make that change. Thanks for the information.